### ISSUE TITLE
smudge filter lfs failed - OSX High Sierra Issue?

### ISSUE THREAD

**Issue by rxmarcus** [git-lfs/git-lfs#2754]
All of the sudden I'm having an issue where when I try to switch branches, I'm getting an error about "smudge filter lfs failed".

I've been reading old posts about this issue and have found no solutions. On my end, all I can think is that I did upgrade from OSX Sierra to High Sierra recently. (No idea if this could be an issue)

I've tried deleting and making new SSH keys, using githubs personal tokens, cloning a fresh copy of the repo, etc. No matter what I do, when I clone a new copy of the repo and try to change branches, I get this error in regards to LFS files.

Here is the log file I'm provided:

> git-lfs/2.0.2 (GitHub; darwin amd64; go 1.8.1)
git version 2.10.1
$ git-lfs smudge -- Assets/Prefabs/Something/Object.fbx
Error downloading object: Assets/Prefabs/Something/Object.fbx (61fc20dee75e368982d950be5315081f72256f3d6c9915abdad3f10f36843f5b)
Smudge error: Error downloading Assets/Prefabs/Something/Object.fbx (61fc20dee75e368982d950be5315081f72256f3d6c9915abdad3f10f36843f5b): batch response: Git credentials for (removed by me) not found.
github.com/git-lfs/git-lfs/errors.newWrappedError
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/errors/types.go:166
github.com/git-lfs/git-lfs/errors.NewSmudgeError
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/errors/types.go:252
github.com/git-lfs/git-lfs/lfs.PointerSmudge
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/lfs/pointer_smudge.go:68
github.com/git-lfs/git-lfs/lfs.(*Pointer).Smudge
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/lfs/pointer.go:64
github.com/git-lfs/git-lfs/commands.smudge
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/commands/command_smudge.go:63
github.com/git-lfs/git-lfs/commands.smudgeCommand
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/commands/command_smudge.go:91
github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra.(*Command).execute
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra/command.go:477
github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra.(*Command).Execute
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra/command.go:551
github.com/git-lfs/git-lfs/commands.Run
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/commands/run.go:68
main.main
	/Users/steve/projects/Go/src/github.com/git-lfs/git-lfs/git-lfs.go:35
runtime.main
	/usr/local/Cellar/go/1.8.1/libexec/src/runtime/proc.go:185
runtime.goexit
	/usr/local/Cellar/go/1.8.1/libexec/src/runtime/asm_amd64.s:2197
ENV:
LocalWorkingDir=/Users/username/ProjectName
LocalGitDir=/Users/username/ProjectName/.git
LocalGitStorageDir=/Users/username/ProjectName/.git
LocalMediaDir=/Users/username/ProjectName/.git/lfs/objects
LocalReferenceDir=
TempDir=/Users/username/ProjectName/.git/lfs/tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
AccessDownload=basic
AccessUpload=basic
DownloadTransfers=basic,bitbucket-media-api
UploadTransfers=basic,bitbucket-media-api
GIT_DIR=.git
GIT_CONFIG_PARAMETERS='color.branch=false' 'color.diff=false' 'color.status=false' 'diff.mnemonicprefix=false' 'core.quotepath=false' 'credential.helper=sourcetree' 'lfs.customtransfer.bitbucket-media-api.path=/Applications/Sourcetree.app/Contents/Resources/git_local/git-lfs/git-lfs-bitbucket-media-api'
GIT_PREFIX=
GIT_PAGER=cat
GIT_EXEC_PATH=/Applications/Sourcetree.app/Contents/Resources/git_local/libexec/git-core

Any ideas or direction would be greatly appreciated. Been stuck on this all day.... :(

**Comment by ttaylorr** [comment]
Hey @rxmarcus, thanks for opening this issue -- sorry that you're having trouble! Here are a few things that I can think of:

1. The error is that your credentials aren't found, which means that this could be a credential helper issue. On macOS, this is usually stored in your Keychain.app, so perhaps something could be going on there.
2. Make sure that Git and Git LFS are up-to-date after upgrading macOS. The current latest versions of both are v2.15.0, and v2.3.4, respectively.

Can you make sure that everything is OK with your credential helper and that you're running the latest versions of Git and Git LFS and let me know if the issue persists?

**Comment by rxmarcus** [comment]
So I seem to have resolved my issue.

I updated to the latest Git and Git LFS versions as you suggested, that did not resolve the problem. I cleaned out all entries for Git inside the Mac OSX keychain manager, that didn't work.

Finally, I cleared those out again, and this time when I setup my Account to GitHub inside of Source Tree I used HTTPS rather than SSH. I read a post somewhere of someone saying that Git LFS doesn't support SSH? I don't know if that is still true today, but after connecting my account via HTTPS instead I was able to clone and pull down my LFS content with no issues.

Thanks for your help @ttaylorr and I hope this helps someone else out in the future.
